org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/BoundsComputer.java

```diff
@@ -90,22 +113,55 @@ private TupleSet computeLowerBound(List<Tuple> atoms, final PrimSig sig) throws
             lower.addAll(computeLowerBound(atoms, c));
         TupleSet upper = lower.clone();
         boolean isExact = sc.isExact(sig);
-        if (isExact || sig.isTopLevel())
-            for (n = n - upper.size(); n > 0; n--) {
-                Tuple atom = atoms.remove(atoms.size() - 1);
-                // If MUST<SCOPE and s is exact, then add fresh atoms to both
-                // LOWERBOUND and UPPERBOUND.
-                // If MUST<SCOPE and s is inexact but toplevel, then add fresh
-                // atoms to the UPPERBOUND.
-                if (isExact)
+        boolean hasLower = sc.hasLower(sig);
+
+        if (hasLower) 
```

Comment:
please add curly braces around the `then` and `else` clauses of this `if`

Reply:
fixed.